File: leetcode/majority-element.cpp
#include <bits/stdc++.h>
using namespace std;

class Solution{
public:
    int majorityElement(vector<int>& nums){
        int current = nums[0];
        int counter = 0;
        int n = nums.size();
        for(int i =0;i<n;++i){
            if(counter==0)current=nums[i];
            if(nums[i]==current)++counter;
            else --counter;
        }
        return current;
    }
};

File: leetcode/majority-element.py
class Solution:
    def majorityElement(self, nums: List[int]) -> int:
        n = len(nums)
        current = nums[0]
        counter = 0
        for x in range(n):
            if(counter==0):
                current = nums[x]
            if(nums[x]==current):
                counter+=1
            else:
                counter-=1
        return current

        